lib/kotkowo_web/components/cards.ex

```diff
@@ -63,13 +63,31 @@ defmodule KotkowoWeb.Components.Cards do
             ])
           }
         />
-        <a
+        <div
           :if={@share_href != nil}
-          href={@share_href}
-          class="absolute right-3 top-3 bg-white w-6 lg:w-10 h-6 lg:h-10 rounded-full opacity-60 flex"
+          x-data={"{shown: false, share_data: {
+          url: '#{@share_href}',
+          text: 'placeholder text',
+          title: 'placeholder title',
+        }}"}
         >
-          <.icon name="share" class="w-3 lg:w-5 h-3 lg:h-5 m-auto" />
-        </a>
+          <div
+            class="absolute right-3 top-3 bg-white w-6 lg:w-10 h-6 lg:h-10 rounded-full opacity-60 flex"
+            x-on:click="
+            if (navigator.canShare(share_data)) {
+              navigator.share(share_data).catch((_error) => {
+            shown = true;
+            })
+            } else {
+            shown = true;
+
+            }
+            "
```

Comment:
this could be moved to JS module later(https://alpinejs.dev/globals/alpine-data#registering-from-a-bundle).
fine for now.